Jmeter Integration with Custom Plugin (#8)
Azure-Samples/azure-load-testing-samples: jmeter-custom-plugin/README.md

Author: Update please check

---
page_type: sample
languages:
- java
products:

> - jmeter Custom Plugin.

Reviewer: Please remove, as this is not an official product in the metadata system.

Author: Done

Reviewer: Please remove the line for jmeter custom plugin. This is not an official product in the metadata system

- azure-load-testing
urlFragment: azure-load-testing-jmeter-custom-plugin

> description: Create and Integrate Custom Plugin for JMeter in Azure Load Testing

Reviewer: ```suggestion
description: Create and integrate a custom plugin for JMeter in Azure Load Testing
```

Author: Done

---

> # Create Custom plugin and Integrate in Jmeter for Azure Load Testing

Reviewer: ```suggestion
# Create and integrate a custom plugin for JMeter in Azure Load Testing
```

Author: resolved

> As .Jar(Java Archive) can be easily integrated to Jmeter. Custom Plugin can be created in Java project and all functions can be called straightly using JSR223Sample or Beanshell sampler.

Reviewer: ```suggestion
This sample demonstrates how to create a custom JMeter plugin and use it with Azure Load Testing. Package the JMeter plugin source code in a JAR (Java Archive) file, and invoke the functions from a script sampler in the JMeter test script (JMX).
```

Author: Done

> In this sample we will talking about steps for creating custom plugin

Reviewer: ```suggestion
```

## Prerequisites

> 

Author: Done

> - An Azure Load Testing resource. An Azure Load Testing resource. To create a Load Testing resource, see [Create and run a load test](https://learn.microsoft.com/azure/load-testing/quickstart-create-and-run-load-test#create-an-azure-load-testing-resource).

Author: Done

> - Creating Custom Plugin and integrating it with JMeter [Create and Integrate custom plugin](https://techcommunity.microsoft.com/t5/apps-on-azure-blog/custom-plugin-creation-and-integration-with-jmeter-and-azure/ba-p/3745773).

Author: Done

## Run this sample

1. Download the Sample Java Maven project - **codepackage**
2. Using Maven create a JAR using the command 
mvn clean compile assembly:single
Note: Execute this command on the location having .pom file present.
3. Link the new JAR in the JMeter Suite - Click on browse and import jar.
Note Use the name of the JAR file instead of using the complete path of the JAR file 
for example. in our case we have name as sampleexample-1.0-jar-with-dependencies.jar
4. Create a BeanShell or JSR223 Sampler in your JMeter Suite and import the package after this create an object of class and call the functions.
5. Upload the .jmx file and .jar file in Azure Load Testing or in ADO pipeline and perform the execution

> 

Author: Done

> 

Reviewer: Can you keep some instructions about what is in this repo and how people can use it? You need to provide information about what is included and how people can use it, don't just defer people to a blogpost for all information.
